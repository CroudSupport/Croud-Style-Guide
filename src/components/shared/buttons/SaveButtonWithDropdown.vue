
<script>
    import Save from './SaveButton'
    import DropdownButton from './DropdownButton'

    /**
    * Combination of Save Button atom and Dropdown Button atom to provide additional options to the Save Button
    *
    * *This is a functional component which has no state so does not need to define any props*
    * @see https://vuejs.org/v2/guide/render-function.html#Functional-Components
    * @see https://croudsupport.github.io/Croud-Style-Guide/technical/#!/croud-save-button
    * @see https://croudsupport.github.io/Croud-Style-Guide/technical/#!/croud-dropdown-button
    *
    * @example ./croud-save-button-with-dropdown.md
    */
    export default {
        functional: true,

        name: 'croud-save-button-with-dropdown',

        render(createElement, context) {
            return createElement('div',
                {
                    class: 'ui blue buttons',
                },
                [
                    createElement(Save, {
                        props: context.props,
                        on: context.listeners,
                    }),
                    createElement(DropdownButton, {
                        props: context.props,
                    }),
                ],
        )
        },
    }
</script>

<style scoped>

</style>

