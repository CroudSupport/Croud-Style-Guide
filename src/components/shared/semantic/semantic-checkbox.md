### semantic-checkbox
For our checkboxes, we use the semantic-checkbox component from the [croud-vue-semantic](https://github.com/CroudSupport/vue-semantic) library. Below are a few examples of semantic-checkbox in action. Click [here](http://croudsupport.github.io/vue-semantic/#semantic-checkbox) for full component docs.

### Basic usage

    <semantic-checkbox v-model="checkbox" label="🍕 Pizza" />

### Styling
Pass through the **type** prop to change the style of checkbo

    <div class="ui form">
        <div class="ui field">
            <semantic-checkbox v-model="checkbox" label="Checkbox" />
        </div>
        <div class="ui field">
            <semantic-checkbox v-model="checkbox" label="Slider" type="slider"/>
        </div>
        <div class="ui field">
            <semantic-checkbox v-model="checkbox" label="Toggle" type="toggle" />
        </div>
    </div>

### Disabled

    <semantic-checkbox label="🍕 Pizza" type="toggle" :disabled="true"/>
