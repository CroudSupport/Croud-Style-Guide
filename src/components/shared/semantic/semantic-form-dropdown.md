### semantic-form-dropdown
For our form dropdowns, we use the semantic-form-dropdown component from the [croud-vue-semantic](https://github.com/croudtech/vue-semantic) library. Below are a few examples of semantic-form-dropdown in action. Click [here](http://croudtech.github.io/vue-semantic/#semantic-form-dropdown) for full component docs.

### Basic usage

    <semantic-form-dropdown v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]"></semantic-form-dropdown>

### Styling
You can use the props to change the styling of the dropdown

    <div class="ui form">
        <div class="field">
            <label>Fluid</label>
            <semantic-form-dropdown fluid v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
        </div>

        <div class="three fields">
            <div class="field">
                <label>Search</label>
                <semantic-form-dropdown search v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
            </div>
            <div class="field">
                <label>Multiple</label>
                <semantic-form-dropdown multiple v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
            </div>
            <div class="field">
                <label>Disabled</label>
                <semantic-form-dropdown :disabled="true" v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
            </div>
        </div>

        <div class="inline field">
            <label>Inline</label>
            <semantic-form-dropdown :inline="true" v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
        </div>
    </div>

### Using slots
The **default** slot allows you to add custom options to the top of the dropdown

    <semantic-form-dropdown v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]">
        <div class="item" v-data="3">
            <b>Baz</b>
        </div>
    </semantic-form-dropdown>

### Handle Events
This component emits the **dropdown-selected** event when an item is selected.

    <semantic-form-dropdown :value="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" @dropdown-selected="$toasted.success('TaDa!!!')"></semantic-form-dropdown>
