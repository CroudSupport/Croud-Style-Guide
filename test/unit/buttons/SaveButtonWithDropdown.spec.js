
import Vue from 'vue'
import SaveButtonWithDropdown from '../../../src/components/shared/buttons/SaveButtonWithDropdown'
import '../../../semantic/dist/semantic'

const Constructor = Vue.extend({
    props: {
        loading: {},
    },
    render(h) {
        return h(SaveButtonWithDropdown, {
            props: {
                loading: this.loading,
                menu: [
                    { title: 'foo', action: () => console.log('foo') },
                    { title: 'bar', action: () => console.log('bar') },
                ],
            },
        })
    },
})

const vm = new Constructor({
    propsData: {
        loading: false,
    },
}).$mount()

describe('SaveButtonWithDropdown', () => {
    describe('should build component', () => {
        it('should match the snapshot', () => {
            expect(vm.$el).toMatchSnapshot()
        })
    })

    it('should handle a loading state', (done) => {
        vm.loading = true
        vm.$nextTick(() => {
            const saveNode = vm.$el.querySelector('.blue.button')
            const dropdownNode = vm.$el.querySelector('.dropdown.button')

            expect(saveNode.classList).toContain('loading')
            expect(dropdownNode.classList).toContain('disabled')
            expect(saveNode.disabled).toBe(true)
            done()
        })
    })
})
