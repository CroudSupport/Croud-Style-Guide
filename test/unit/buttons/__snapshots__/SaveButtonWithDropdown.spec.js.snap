// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveButtonWithDropdown should build component should match the snapshot 1`] = `
<div
  class="ui blue buttons"
>
  <button
    class="ui blue button"
  >
    Save
  </button>
  <div
    class="ui floating dropdown icon button"
    tabindex="0"
  >
    <i
      class="dropdown icon"
    />
    <div
      class="menu"
      tabindex="-1"
    >
      <div
        class="item"
      >
        foo
      </div>
      <div
        class="item"
      >
        bar
      </div>
    </div>
  </div>
</div>
`;
